Include <arpa/inet.h> for htons() and ntohs().

Should be the portable solution specified by man pages.

diff -u -r ../gnutls-3.3.8/src/socket.c ./src/socket.c
--- ../gnutls-3.3.8/src/socket.c	2014-08-23 07:19:59.000000000 +0200
+++ ./src/socket.c	2014-09-19 22:48:54.148437077 +0200
@@ -46,6 +46,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <arpa/inet.h>
 #ifndef _WIN32
 #include <signal.h>
 #endif
